core/src/blocktree.rs

```diff
@@ -3114,6 +3122,40 @@ pub mod tests {
         Blocktree::destroy(&blocktree_path).expect("Expected successful database destruction");
     }
 
+    #[test]
+    fn test_set_root() {
```

Comment:
does this test fail without the changes above?

Reply:
Yeah, it causes the leader_schedule_cache to fail when asking for the leader of the 96th (first slot of the 2nd epoch) slot b/c no root was set in the first 95 slots.